varinfo/var_info.py

```diff
@@ -711,6 +747,9 @@ def _parse_group(self, group: Dataset | Group):
                 full_name_path=variable_path,
             )
 
+            for index, dimensions_name in enumerate(variable.dimensions):
+                self.all_dimensions_sizes[dimensions_name] = variable.shape[index]
```

Comment:
`enumerate` is a good way to populate `VarInfoFromNetCDF4.all_dimension_sizes`, but this class doesn't have this attribute as you declare it in `VarInfoFromDmr.__init__` and then use `super` to add in all the basic stuff from `VarInfoBase`. `all_dimension_sizes` is also not used anywhere in the rest of the `VarInfoFromNetCDF4` or `VariableFromNetCDF4` classes, so I don't think you need these lines (unless I'm missing something).

There are two possible resolutions here:

1) Remove these lines.
2) Add `all_dimension_sizes` to `VarInfoBase.__init__` rather than in `VarInfoFromDmr.__init__`.

Reply:
Going to mark this as resolved. I was getting mixed up by the `VarInfoBase` having the class attribute, but the attribute only being passed to `VariableFromDmr`. Sorry about that!